--- a/src/hotspot/make/solaris/makefiles/gcc.make
+++ b/src/hotspot/make/solaris/makefiles/gcc.make
@@ -117,7 +117,7 @@ endif
 
 
 # Compiler warnings are treated as errors
-WARNINGS_ARE_ERRORS ?= -Werror
+WARNINGS_ARE_ERRORS ?= -Werror -Wno-error=format
 
 # Enable these warnings. See 'info gcc' about details on these options
 WARNING_FLAGS = -Wpointer-arith -Wconversion -Wsign-compare -Wundef -Wformat=2
--- a/src/hotspot/make/linux/makefiles/gcc.make
+++ b/src/hotspot/make/linux/makefiles/gcc.make
@@ -203,7 +203,7 @@ else
 endif
 
 # Compiler warnings are treated as errors
-WARNINGS_ARE_ERRORS ?= -Werror
+WARNINGS_ARE_ERRORS ?= -Werror -Wno-error=format
 
 ifeq ($(USE_CLANG), true)
   # However we need to clean the code up before we can unrestrictedly enable this option with Clang
